pytoda/datasets/annotated_dataset.py

```diff
@@ -94,19 +97,98 @@ def __getitem__(self, index: int) -> AnnotatedData:
 
         Returns:
             AnnotatedData: a tuple containing two torch.tensors,
-                representing respetively: compound token indexes and labels for
-                the current sample.
+                representing respectively: compound token indexes and labels
+                for the current sample.
         """
         # sample selection
-        selected_sample = self.annotated_data_df.iloc[index]
+        selected_sample = self.df.iloc[index]
+        # label
+        labels_tensor = torch.tensor(
+            list(selected_sample[self.labels].values),  # base_dataset: why selecting self.labels here and not for all in __init__? want to change self.labels on the instance?
```

Comment:
it's a conversation starter.

instead of storing self.labels and index using them each time, we could just index these columns of the df in the init.

Except one wants to change self.labels in the instance.
Is this a usecase for the annotated dataset?

Reply:
I don't think that's a common usecase. Would definitely prefer to change it since it should speed up things for wide DFs with few used columns.

Great catch!